### PR TITLE
tests: provide tests for od module

### PR DESCRIPTION
**Pull request by miri64**
Just realized that this module isn't tested anywhere so I provided some tests. It is very likely, that the test script isn't working on non-32-bit, non-little-endian platforms so this could be the perfect example to get some kind of unified test system started ;-).

### CODE REVIEW

**Comment by gebart** [comment]
It's always good to increase the test coverage! Nice!

**Comment by miri64** [comment]
Then is someone willing to review ;-)?

File: tests/od/main.c [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "od.h"
+
+#define CALL(fn)    puts(#fn); fn
+
+static const char short_str[] = "AB";
```
Comment (A-Paul) [inline tests/od/main.c]: Do these strings provide any dedicated special bit patterns, or are they just arbitrary?

Comment (miri64) [inline tests/od/main.c]: Just arbitrary.

Comment (miri64) [inline tests/od/main.c]: Oh well wait: the `long_str` contains bit patterns that have different results in unsigned and signed notation: `0xff`.

Comment (miri64) [inline tests/od/main.c]: `short_str` is only used to test output width vs array width discrepencies.

File: tests/od/main.c [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+int main(void)
+{
+    CALL(od(short_str, sizeof(short_str), OD_WIDTH_DEFAULT, 0));
+
+    CALL(od(long_str, sizeof(long_str), OD_WIDTH_DEFAULT, 0));
+    CALL(od(long_str, sizeof(long_str), 4, 0));
+    CALL(od(long_str, sizeof(long_str), 3, 0));
+    CALL(od(long_str, sizeof(long_str), 8, 0));
+
+
```
Comment (A-Paul) [inline tests/od/main.c]: I'm trying to imagine a pattern for the use of one or two empty lines between the code blocks.
Could this be some reserved space for the comments explaining what is tested by the different "cases"? ;-)

Comment (miri64) [inline tests/od/main.c]: Don't know, but I can add some comments if that is what you want ;-)

**Comment by A-Paul** [comment]
Syntax and code style seem O.K. to me. Built and run on native, but I'm unsure what the expected output is.
The number of test cases is overwhelming and I don't get which different errors or error classes they are covering.

File: tests/od/main.c (hunk 2 above)
Comment (A-Paul) [inline tests/od/main.c]: That's what I intended between the lines. :D

**Comment by miri64** [comment]
The expected output is in [the contained expect script](https://github.com/miri64/RIOT/blob/f97d89b1ab200d92785b77a65c4854b0ebe291f0/tests/od/tests/01-run.py)

**Comment by miri64** [comment]
(at least for 32-bit platforms)

**Comment by miri64** [comment]
(and since `od` does not contain erroring functions it just tests if the output is as expected, no error classes are tested).

File: tests/od/main.c (hunk 2 above)
Comment (miri64) [inline tests/od/main.c]: Done.

**Comment by A-Paul** [comment]
Thanks for the comments. Do you mind to do the same to the response section?

**Comment by miri64** [comment]
> Do you mind to do the same to the response section?

Do you mean the python script?

**Comment by miri64** [comment]
>> Do you mind to do the same to the response section?
>
> Do you mean the python script?

I am just assuming you meant that and did it.

**Comment by A-Paul** [comment]
@miri64,  sorry for being delayed again. Thank your for addressing the comments.
Please squash. If Murdock is green I'll ACK.

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
Murdock is happy :-)

**Pull request merged**